memdb.go

```diff
@@ -95,6 +95,44 @@ func (db *MemDB) Snapshot() *MemDB {
 	return clone
 }
 
+// CreateIndexes is used for create indexes after the database already exists.
+func (db *MemDB) CreateIndexes(table string, schema ...*IndexSchema) error {
+	// lock the writer but create the transaction afterwards
+	db.writer.Lock()
+
+	// create a edge for each index to create
+	root := db.getRoot()
+	for _, indexSchema := range schema {
+		index := iradix.New()
+		path := indexPath(table, indexSchema.Name)
+		root, _, _ = root.Insert(path, index)
+	}
+	db.root = unsafe.Pointer(root)
+
+	// now we can create the transaction
+	transaction := &Txn{
+		db:      db,
+		write:   true,
+		rootTxn: db.getRoot().Txn(),
+	}
+	defer transaction.Abort()
+
+	// create the requested indexes
+	if err := transaction.CreateIndexes(table, schema...); err != nil {
+		return err
+	}
+
+	transaction.Commit()
+
+	// copy the new created indexes into our schema
+	tableSchema := db.schema.Tables[table]
+	for _, indexSchema := range schema {
+		tableSchema.Indexes[indexSchema.Name] = indexSchema
+	}
```

Comment:
Hmmm this seems like a race footgun that we should at least document in the doc comment for this method.

I think it's true that today we never expect `db.schema` to change after initialization therefore readers will read from it without taking the `db.writer.Lock` which would be a data race if there were any readers concurrent with this call.

I think that's probably OK for it's intended usage and adding a lock to every read would be too invasive and impact performance for all users, but we should at least point out the requirement in the doc comments that this MUST NOT be called concurrently with any reads. For it's intended purpose that would probably be a logical bug anyway since the indexes the readers presumably need wouldn't exist yet but it might not be obvious to other library users!